store/src/java/com/zimbra/cs/util/JMSession.java

```diff
@@ -351,6 +353,43 @@ private static Properties getJavaMailSessionProperties(Server server, Domain dom
         return props;
     }
 
+    private static Properties configureStartTls (Properties props, Server server, Domain domain) {
+        SmtpStartTlsMode startTlsMode = null;
+        String sslTrustedHosts = null;
+
+        if (domain != null) {
+            startTlsMode = domain.getSmtpStartTlsMode();
+            sslTrustedHosts = domain.getSmtpStartTlsTrustedHosts();
+        }
+        if (startTlsMode == null) {
+            startTlsMode = server.getSmtpStartTlsMode();
+        }
+        if (sslTrustedHosts == null) {
+            sslTrustedHosts = server.getSmtpStartTlsTrustedHosts();
+        }
+
+        if (startTlsMode.isOff()) {
+            props.setProperty("mail.smtp.starttls.enable", "false");
+        } else if (startTlsMode.isOnly()) {
+            props.setProperty("mail.smtp.starttls.enable", "true");
+            props.setProperty("mail.smtp.starttls.required", "true");
+            if (sslTrustedHosts != null) {
+                props.setProperty("mail.smtp.ssl.trust", sslTrustedHosts);
+            }
+        } else {
+            if (!startTlsMode.isOn()) {
+                ZimbraLog.smtp.warn("invalid value configured for %s. fallback to \"on\".", Provisioning.A_zimbraSmtpStartTlsMode);
```

Comment:
nit: change `"invalid value configured for %s. fallback to \"on\"."` to ...
something more consistent with other parts of the product (ha - if only we were consistent :) ) here's one way we've logged similar issues:
`.../AttributeInfo.java:                    ZimbraLog.misc.warn("Invalid value '%s' for property %s of attribute %s.  Defaulting to %d."`
so perhaps:
`ZimbraLog.smtp.warn("Invalid value '%s' for %s. Defaulting to 'on'.", sslTrustedHosts, Provisioning.A_zimbraSmtpStartTlsMode);`
or just
`ZimbraLog.smtp.warn("Invalid value for %s. Defaulting to 'on'.", Provisioning.A_zimbraSmtpStartTlsMode);`

Reply:
I really appreciate your help. Will be fixed in next commit.